internal/datagen/generator.go

```diff
@@ -58,8 +58,10 @@ func (g *Generator) nextSlice() []byte {
 	if g.buf == nil {
 		// Allocate buffer with extra tail for sliding and populate it with random bytes
 		g.buf = make([]byte, g.size+TailSize)
-		//nolint:staticcheck
-		rand.Read(g.buf) // Per docs, err is always nil here
+		_, err := rand.Read(g.buf)
+		if err != nil {
+			panic(err)
```

Comment:
And `panic` is ok here, I think. At least not worth additional code for "proper" handling, the tool will fail completely anyway in this case.